### PR TITLE
Add type hints to hyperframe

### PR DESCRIPTION
**Pull request by sethmlarson**
I'm unsure about the types within `flags.py`, let me know what you think.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -4,4 +4,5 @@ graft test
 prune docs/build
 prune test/http2-frame-test-case
 include README.rst LICENSE CHANGELOG.rst CONTRIBUTORS.rst tox.ini .gitmodules
+include src/hyperframe/py.typed
```
Comment (Kriechi) [inline MANIFEST.in:7]: Does this have the same effect as adding it to `package_data` in the setup call?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,7 +6,7 @@ dev
 
 **API Changes (Backward Incompatible)**
 
--
+- Added type hints.
```
Comment (Kriechi) [inline CHANGELOG.rst:9]: Are there actually backward incompatible changes?
The next major release might be a long time away...

**Review by Kriechi — APPROVED** [review]

**Comment by Kriechi** [comment]
Tests are failing, two minor questions/comments, but otherwise 👍

File: CHANGELOG.rst (hunk 2 above)
Comment (sethmlarson) [reply CHANGELOG.rst:9]: Woops! No these aren't breaking changes, I read this as "Backwards compatible", doh.

File: MANIFEST.in (hunk 1 above)
Comment (sethmlarson) [reply MANIFEST.in:7]: I'll test this, but I think I should add to package_data as well

**Comment by graingert** [comment]
@Kriechi can you publish this to make it easier to add type hints to h2?

**Comment by Kriechi** [comment]
Done. :shipit:

**Comment by graingert** [comment]
Awesome, I think hpack is the last dep now

**Pull request merged**